docs/development/deploying-locally/index.md

```diff
@@ -1,3 +1,50 @@
+
+### Instructions
+
+* (1) Clone the main REANA repository:
+```console
+$ git clone https://github.com/reanahub/reana.git
+$ cd reana
+```
+
+* (2) Create a REANA development setup as follows:
+```console
+$ GITHUB_USER=JOHNDOE make setup clone prefetch
+```
+
+* (3) Build all components:
+```console
+$ make build
+```
+
+
+* (4) Deploy REANA:
+```console
+$ make deploy
+```
+
+Note that, REANA can also be deployed in development mode with code reload and [debugging capabilities](../debugging)
+```
+$ CLUSTER_FLAGS="debug.enabled=true" make deploy
+```
+
+### Makefile variables reference
+
+| Variable              | Description                                                                                         | Default value  |
+|-----------------------|-----------------------------------------------------------------------------------------------------|----------------|
+| `CLUSTER_FLAGS`       | Which [values](https://github.com/reanahub/reana/blob/master/helm/reana/README.md) need to be passed to `helm install`? (e.g. `debug.enabled=true,ui.enable=true`) | - |
+| `DEMO`                | Which [demo example](https://github.com/reanahub?q=reana-demo) to run? (e.g. [`reana-demo-helloworld`](https://github.com/reanahub/reana-demo-helloworld)) | several |
+| `EXCLUDE_COMPONENTS`  | Which REANA components should be excluded from the build? [e.g. `reana-ui,r-m-broker`]                | -              |
+| `GITHUB_USER`         | Which GitHub user account to use for cloning the REANA reposistories?                               | `anonymous`    |
+| `INSTANCE_NAME`       | Which name/prefix to use for your REANA instance?                                                   | `reana`        |
+| `MINIKUBE_CPUS`       | How many CPUs to allocate for Minikube?                                                             | `2`            |
+| `MINIKUBE_DISKSIZE`   | How much disk size to allocate for Minikube?                                                        | `40g`          |
+| `MINIKUBE_DRIVER`     | Which [VM driver](https://minikube.sigs.k8s.io/docs/reference/drivers/) to use for Minikube?        | `virtualbox`   |
+| `MINIKUBE_KUBERNETES` | Which [Kubernetes version](https://github.com/kubernetes/kubernetes/releases) to use with Minikube? | `v1.16.3`      |
+| `MINIKUBE_MEMORY`     | How much memory to allocate for Minikube?                                                           | `3072`         |
+| `SERVER_URL`          | Setting a customized REANA Server hostname? (e.g. `https://example.org`)                            | https://[`` `minikube ip` ``](https://minikube.sigs.k8s.io/docs/reference/commands/ip/) |
+| `TIMECHECK`           | Checking frequency in seconds when bringing cluster up and down?                                    | `5`            |
+| `TIMEOUT`             | Maximum timeout to wait when bringing cluster up and down?                                          | `300`          |
```

Comment:
Why not to print the whole `make` help output? It would give nice recipes (Example 1-8) as well... Just musing out loud.

Reply:
I just think a table reads better, plus it can contain links (see link to know what you can pass to `CLUSTER_FLAGS`:
![image](https://user-images.githubusercontent.com/9035606/78577674-b93f8d00-782e-11ea-9b23-cf8b4416bc68.png)
I understand it is "more to maintain" but doesn't look like a big deal to me.

Reply:
What I mean is that people will miss recipes Example 1-8... I.e. the examples how these flags are used. They will only see the table.

We can keep two examples for now...